examples/new_route.rs

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+
+use std::net::{Ipv4Addr};
+use netlink_packet_core::{
+    NetlinkHeader, NetlinkMessage, NetlinkPayload, NLM_F_ACK, NLM_F_CREATE,
+    NLM_F_EXCL, NLM_F_REQUEST,
+};
+use netlink_packet_route::{
+    constants::{AF_INET}, route, RtnlMessage, RouteHeader, RouteMessage, RouteFlags,
+    RT_TABLE_MAIN, RTPROT_BOOT, RT_SCOPE_UNIVERSE, RTN_UNICAST,
+};
+use netlink_sys::{protocols::NETLINK_ROUTE, Socket, SocketAddr};
+
+fn main() {
+    let mut socket = Socket::new(NETLINK_ROUTE).unwrap();
+    let _port_number = socket.bind_auto().unwrap().port_number();
+    socket.connect(&SocketAddr::new(0, 0)).unwrap();
+
+    let mut route_msg_hdr = RouteHeader::default();
+    route_msg_hdr.address_family = AF_INET as u8;
+    route_msg_hdr.table = RT_TABLE_MAIN;
+    route_msg_hdr.scope = RT_SCOPE_UNIVERSE;
+    route_msg_hdr.protocol = RTPROT_BOOT;
+    route_msg_hdr.kind = RTN_UNICAST;
+    route_msg_hdr.source_prefix_length = 0;
+    route_msg_hdr.destination_prefix_length = 0;
+    route_msg_hdr.flags = RouteFlags::RTNH_F_ONLINK;
+
+    let mut route_msg = RouteMessage::default();
+    route_msg.header = route_msg_hdr;
+    route_msg.nlas = vec![
+        route::Nla::Oif(2),
+        route::Nla::Gateway(
+            "169.254.1.1".parse::<Ipv4Addr>().unwrap().octets().to_vec()),
+        route::Nla::Destination(
+            "0.0.0.0".parse::<Ipv4Addr>().unwrap().octets().to_vec()),
```

Comment:
Instead of gateway, maybe a /32 route could be less destructive.

Reply:
ok, use `1.1.1.1 via 169.254.1.1 dev lo onlink`.